Form.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
+
+import './Form.css';
+
+class Form extends Component {
+  state = {
+    form: {
+      firstName: '',
+      lastName: '',
+      email: '',
+      password: '',
+    },
+  };
+
+
+  handleFirstNameChange = (e) => {
+    this.setState({
+      form: {
+        firstName: e.target.value,
+      },
+    });
+  };
+
+  handleLastNameChange = (e) => {
+    this.setState({
+      form: {
+        lastName: e.target.value,
+      },
+    });
+  };
+
+  handleEmailChange = (e) => {
+    this.setState({
+      form: {
+        email: e.target.value,
+      },
+    });
+  };
+
+  handlePasswordChange = (e) => {
+    this.setState({
+      form: {
+        password: e.target.value,
+      },
+    });
+  };
```

Comment:
There is a lot of repetition here. A single function that can set any key in `state.form` might be easier to read and maintain.

However, there is a trade-off to be considered between avoiding repetition and tightly coupling the handlers of the individual inputs. If the behaviour of one of the inputs diverges from the others, we might have to violate the open/closed principle to unpick them, which could inadvertently modify the behaviour of existing UI elements

I would consider creating a change handler that understands the parameters sent by an `input[type=text]` and uses those parameters to call a separate function that knows how to modify the component state. This might seem somewhat redundant initially but should make subsequent modifications quicker and safer.